impl.hpp

```diff
@@ -1163,13 +1243,25 @@ bool Client::ndt7_upload() noexcept {
         NDT7_ENUM_TCP_INFO
 #undef XX
       }
+
+      // Calculate retransmission rate.
+      try {
+        nlohmann::json tcpinfo_json = measurement["TCPInfo"];
+        double bytes_retrans = (double) tcpinfo_json["TcpiBytesRetrans"].get<int64_t>();
+        double bytes_sent = (double) tcpinfo_json["TcpiBytesSent"].get<int64_t>();
+        summary_.upload_retrans = bytes_retrans / bytes_sent;
```

Comment:
My main concern with this diff is that we should avoid accidental division by zero.